packages/webdriverio/src/commands/browser/action.ts

```diff
@@ -0,0 +1,146 @@
+import { KeyAction, PointerAction, WheelAction, ActionType, BaseActionParams } from '../../utils/actions/index.js'
+
+/**
+ * The action command is a low-level interface for providing virtualized device input actions to the web browser.
+ *
+ * In addition to high level commands such like `scrollIntoView`, `doubleClick`, the Actions API provides granular
+ * control over exactly what designated input devices can do. WebdriverIO provides an interface for 3 kinds of input
+ * sources:
+ *
+ * - a key input for keyboard devices
+ * - a pointer input for a mouse, pen or touch devices
+ * - and wheel inputs for scroll wheel devices
+ *
+ * Every chain of action commands has to be completed calling `perform` in order to trigger the set of actions.
+ *
+ * ### Key input source
+ *
+ * A key input source is an input source that is associated with a keyboard-type device. It can be triggered
+ * by using the `key` type parameters. e.g.:
+ *
+ * ```ts
+ * browser.action('key')
+ * ```
+ *
+ * It returns a `KeyAction` object that supports the following actions:
+ *
+ * - `down(value: string)`: generates a key down action
+ * - `up(value: string)`: generates a key up action
+ * - `pause(ms: number)`: indicate that an input source does nothing during a particular tick
+ *
+ * ### Pointer input source
+ *
+ * A pointer input source is an input source that is associated with a pointer-type input device. The type can be
+ * specified when invoking the `action` command, e.g.:
+ *
+ * ```ts
+ * browser.action('key'. {
+ *     parameters: { pointerType: 'mouse' } // "mouse" is default value, also possible: "pen" or "touch"
```

Comment:
Did you mean to add a mouse action here? It feels like we are mixing keyboard and mouse and actions with performActions